Reusable Configurations topic (#118)
mulesoft/docs-design-center: modules/ROOT/pages/fd-reuse-config.adoc

= Create a Connection for Reuse by Yourself or Others
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords:

:imagesdir: ../assets/images

You can reuse a connection in a Flow Designer project instead of adding your authentication information manually. Once a reusable connection is created, it is available for all Flow Designer projects in your business group.

You can reuse connections:

* If you are a system admin, you can enable non-admins to connect to your system using your authentication. This is important if the integration requires administrative system permissions.

* If you are responsible for maintaining the connection, you can ensure that the integration is unaffected when the user who created the Flow Designer project changes their personal password or leaves the organization.

> 

Author: Added a sentence.

== Create a Reusable Connection

> Add a reusable connection when configuring a connector in Flow Designer.

Author: Nope, removing.

. Choose an operation.

. Click *Add connection*.

. Enter a unique and descriptive name that your users can easily recognize.

> . To make this connection available for users in your business group, leave the *share this connection with my business group* checkbox checked. To maintain security, only you as the creator of a shared connection can view your authentication credentials.

Author: Fixed.

. If you don’t need to share this connection with other users, uncheck the box.

> . Select a connection type and fill in the necessary information.

Author: Fixed.

. Validate that this connection works by clicking *Test*.

. Click *Save* in the connection section, and click *Save* again in the reusable configuration section.

== Use a Reusable Connection Created in Another Project

> Once a reusable connection is created, it is available for all Flow Designer projects.

Author: Fixed.

. Choose a connector and operation.

> . View a list of reusable connections. To help you choose the right connection the date the reusable connector was created is visible under the connection.

Author: Fixed. The lead-in sentence says that once a reuseable connection is created, then all the reusable ones are viewable. It's literally the first thing the user sees.

. Select the appropriate radio button for the connection you intend to use.

. Click *Test* to make sure that the credentials are valid.

> . Click *Save* after the test succeeds.

Author: Would it be more accurate to say, click save when the test succeeds?

> == Change a reusable connection

Author: But I always thought that change implies add, edit, and delete where edit only implies change existing.

> When your credentials change, you can edit or delete the shared connection you created so that all projects dependent upon it continue to work.

Reviewer: ```suggestion
When your credentials change, you can edit or delete the shared connection you created so that all projects dependent upon it continue to work:
```

Author: Fixed.

> . Open the reusable configuration section of your connector. You see a list of connections. Select one and click the three-dots icon.

Reviewer: ```suggestion
. Open the reusable configuration section of your connector. 
. In the list of connections, select one and click the ... icon.
```
There are several names for the icons that have three horizontal or three vertical dots.
If they are horizontal, you could call this an "ellipsis" icon. 
I've also heard it called a Twinkie!
Some people call three vertical dots the "hamburger" icon.

Author: would love a standard name.  I am using what Robert used in FD to be consistent.

> . If you want to remove the connection, select *Delete*. You must have another connection available to use in order to delete your reusable connection.

Reviewer: ```suggestion
. If you want to remove the connection, select *Delete*. 
You must have another connection available before you can delete your reusable connection.
```

Author: Fixed.

. If you want to edit the connection, select *Edit* and make your changes.

. Click *Save* in the connection section, and click *Save* again in the reusable configuration section.

> This immediately updates all Flow Designer projects being designed, but not Mule runtime apps already deployed to CloudHub.

Reviewer: ```suggestion
Completing this task immediately updates all Flow Designer projects being designed; it does not update Mule runtime apps already deployed to CloudHub.
```

Author: Fixed.

= Create a Connection for Reuse by Yourself or Others
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords:

:imagesdir: ../assets/images

> You can create, change and reuse a connection in a Flow Designer project instead of adding your authentication information manually. Once a reusable connection is created, it is available for all Flow Designer projects.

Reviewer 2: Suggest deleting "create, change and".  "Reuse" is the new feature that we want to call attention to.

Author: Fixed.

> You can create, change and reuse a connection in a Flow Designer project instead of adding your authentication information manually. Once a reusable connection is created, it is available for all Flow Designer projects.

Reviewer 2: suggest changing the end to add clarity: "for all Flow Designer projects in your business group."

Author: Fixed.

You can reuse connections:

> * If you are a system admin, you can enable non-admins to connect to your system using administrative authentication. This is important if the integration requires administrative system permissions.

Reviewer 2: I have never heard the term "administrative authentication".  I think people ordinarily use "administrative credentials" or "an administrative connection."

Author: I was using the term authentication because it looks like it is used throughout most of the FD docs. What about 'your authentication'?

* If you are responsible for maintaining the connection, you can ensure that the integration is unaffected when the user who created the Flow Designer project changes their personal password or leaves the organization.

== Create a Reusable Connection

> Adding a reusable connection is available to you when configuring a connector in Flow Designer.

Reviewer 2: The "available to you" language seems awkward to me.  I suggest "You can add a reusable connection when configuring a connector..." or "Add a connection when configuring..."

Author: Fixed.

. Once you are viewing the Flow Designer canvas, click *+* to add a card.  A list of the available connectors displays within the card.

> . In the Search field, type the name of the connector and select it. The connection section of the connector displays a list of operations that you can use to trigger Mule applications with the connector.

Reviewer 2: Do you think this whole thing can be shortened to "Choose an operation."?

Author: Ha! Yeah, that's a win in the minimalist category.

. Click *Add connection*.

. Enter a unique and descriptive name that your users can easily recognize.

. To make this connection available for users in your business group, leave the *share this connection with my business group* checkbox checked. To maintain security, only you as the creator of a shared connection can view your authentication credentials.

. If you don’t need to share this connection with other users, uncheck the box.

. Select a connection type and fill in the necessary information.

. Validate that this connection works by clicking *Test*.

> . In order to save your connection, you’ll need to click *Save* twice.

Reviewer 2: I suggest rewording so it sounds less like a bug.  I suggest "Click save in the connection section, and click save again in the reusable configuration section."

Author: I think this is being fixed, right?

Author: I have added your suggested change.

== Use a Reusable Connection Created in Another Project

Once a reusable connection is created, it is available for all Flow Designer projects.

. Choose a connector and operation.

. View a list of reusable connections. To help you choose the right connection the date the reusable connector was created is visible under the connection.

. Select the appropriate radio button for the connection you intend to use.

. Click *Test* to make sure that the credentials are valid.

. Click *Save* after the test succeeds.

== Change a reusable connection

> When your credentials change, you can edit or delete the shared connection you created so that all apps dependent upon it continue to work.

Reviewer 2: Technically, it should be "so that all projects" instead of "so that all apps".  A Flow Designer project only becomes an app once it is pushed to production or exported.

Author: Thanks for this. I see why it's incorrect in this instance.

Author: Fixed.

> . Select the connector and click the three-dots icon under Reusable Configurations. If you have the connector open, select a connector under Connection.

Reviewer 2: I suggest clarifying the clicks, perhaps with the following language: "Open the reusable configuration section of your connector.  You see a list of connections.  Select one and click the three-dots icon."

Author: I agree, much better. Fixed.

. If you want to remove the connection, select *Delete*. You must have another connection available to use in order to delete your reusable connection.

. If you want to edit the connection, select *Edit* and make your changes.

> . To have your changes take effect, click *Save* twice.

Reviewer 2: Instead of "click *Save* twice" I recommend "after saving your connection, click save in the reusable configuration section of the connector."  It makes it seem less like a bug.

Author: Fixed.
